extensions/amp-story/1.0/amp-story.js

```diff
@@ -1851,6 +1852,25 @@ export class AmpStory extends AMP.BaseElement {
     }));
   }
 
+
+  /**
+   * @param {!AmpStoryPage} page The page whose CTA anchor tags should be
+   *     upgraded.
+   * @param {number} pageIndex The index of the page.
+   * @private
+   */
+  upgradeCtaAnchorTagsForTracking_(page, pageIndex) {
+    const pageId = page.element.id;
+    const ctaAnchorEls =
+        scopedQuerySelectorAll(page.element, 'amp-story-cta-layer a');
+
+    Array.prototype.forEach.call(ctaAnchorEls, ctaAnchorEl => {
+      ctaAnchorEl.setAttribute('data-vars-story-page-id', pageId);
+      ctaAnchorEl.setAttribute('data-vars-story-page-index', pageIndex);
+    });
+  }
```

Comment:
Should we wrap these read/write operations in a measure/mutate?
This read should not trigger any repaint so it's not a big deal, but it might allow the runtime to do this work in another frame, so it doesn't block this current frame during the layout phase

Reply:
Good catch!  Done.